titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java

```diff
@@ -340,8 +342,13 @@ Completable doAgentScaling() {
                                                                      long elapsed) {
         return instancesForActiveInstanceGroups.entrySet().stream()
                 .flatMap(e -> e.getValue().stream().filter(i -> {
-                    InstanceOverrideStatus overrideStatus = i.getOverrideStatus();
-                    return overrideStatus.getState() == InstanceOverrideState.Removable && hasTimeElapsed(overrideStatus.getTimestamp(), finish, elapsed);
+                    String removableTimestampValue = i.getAttributes().get(AgentAttributes.REMOVABLE);
```

Comment:
A help function in the `AgentAttributes` class? This fetch and check logic will be executed in many other places.